Add a public message for PSU-only works

nicktiberi opened the pull request:
Fixes #662.

When a public user views a work that is PSU-only, a message appears indicating that the files are not viewable.

### Embargoed Work
<img width="1472" alt="image" src="https://user-images.githubusercontent.com/639920/117488610-9f451200-af3a-11eb-9a9d-4e3a4408ad04.png">

### Embargoed, PSU-Only Work
<img width="1472" alt="image" src="https://user-images.githubusercontent.com/639920/117488598-9bb18b00-af3a-11eb-9c5d-a3b746c4ff2c.png">

### PSU-Only Work
<img width="1472" alt="Screen Shot 2021-05-07 at 1 45 19 PM" src="https://user-images.githubusercontent.com/639920/117488550-8f2d3280-af3a-11eb-8448-7197d22037cf.png">

awead commented:
I've updated this based on suggestions from @olendorf and used i18n for the file set titles and error messages.

nicktiberi committed:
Add a public message for PSU-only works
When a public user views a work that is PSU-only, a message appears indicating that the files are not viewable.

nicktiberi commented on spec/components/files_visibility_detail_component_spec.rb:
```diff
@@ -0,0 +1,84 @@
+        expect(content).to include('files will be accessible to users logged-in with a Penn State Access ID.')
+      end
+    end
+  end
```
@awead - this test is failing, probably because I'm not initializing the work and/or work version properly. Am I missing something obvious here?

nicktiberi commented on spec/components/files_visibility_detail_component_spec.rb:
```diff
@@ -0,0 +1,84 @@
+        expect(content).to include('Penn State Only')
+        expect(content).to include('Files are only accessible to users logged-in with a Penn State Access ID.')
+      end
+    end
```
@awead - this test is also failing, probably for the same reason as the test above.

awead commented on app/components/files_visibility_detail_component.rb:
```diff
@@ -0,0 +1,37 @@
+      if unauthorized? then 'embargo_unauthorized' else 'embargo' end
+    elsif unauthorized?
+      'unauthorized'
+    end
```
The nested `if` is a bit dodgy. I have two alternatives:

Snazzy:
``` ruby
    keys = Array.new                                                                                                       
    keys << 'embargo' if embargoed?                                                                                        
    keys << 'unauthorized' if unauthorized?                                                                                
    keys.join('_')
```

Safe:
``` ruby                                                                                                                                                                                                                  
    if unauthorized? && embargoed?                                                                                         
      'embargo_unauthorized'                                                                                               
    elsif embargoed?                                                                                                       
      'embargo'                                                                                                            
    elsif unauthorized?                                                                                                    
      'unauthorized'                                                                                                       
    end

nicktiberi replied:
Yeah, I totally agree. IMO your "safe" option is the most easily readable out of the 3.

awead commented on app/views/shared/_work_version_files.html.erb:
```diff
@@ -1,4 +1,4 @@
-<%= render(EmbargoDetailComponent.new(work_version: work_version)) %>
+<%= render(FilesVisibilityDetailComponent.new(work_version: work_version)) %>
 <%= render(WithdrawnDetailComponent.new(work_version: work_version)) %>
```
Were we combining the withdrawn component? I can't remember... if the work is withdrawn _and_ embargoed/PSU, we'd see two alert boxes. Is that what we want? I apologize if we already covered this.

nicktiberi replied:
@awead - no worries! That was our original plan, but after talking with Seth in Slack on Friday, he preferred to keep it as two separate components and was OK with 2 alert boxes:

> Seth Erickson 9:50
> it might make sense to treat withdrawn in a separate component
> 
> Nick Tiberi  9:51 AM
> ok. the current implementation is that withdrawn and embargoed are 2 separate components, and both could possibly display at the same time. and there is no component for “penn state only”
> 
> Seth Erickson  9:59 AM
> right. so one component for penn state only & embargo, and separate component for withdrawn. This makes sense to me because they relate to slightly different things (works and versions, respectively)

awead replied:
Cool, thanks!
